server/src/test/java/org/elasticsearch/cluster/service/MasterServiceTests.java

```diff
@@ -2121,6 +2121,56 @@ public void onFailure(Exception e) {
         }
     }
 
+    public void testTimeoutRejectionBehaviourAtSubmission() {
+
+        final var actionCount = new AtomicInteger();
+
+        final var deterministicTaskQueue = new DeterministicTaskQueue();
+        final var threadPool = deterministicTaskQueue.getThreadPool(r -> {
```

Comment:
So this is setting up the DeterministicTaskQueue's ThreadPool to execute this code before any Runnable? Could we have a comment about what this is doing? Maybe something like,

"The MasterService wraps update tasks in TaskTimeoutHandlers. Set up the thread pool to check for TaskTimeoutHandler Runnables, and artificially inject an error, but otherwise allow anything else to run as normal."

Reply:
sure, is 5eb4d290cb364682b08a9154804dfbcaf4e3761e ok?

Reply:
I think name dropping MasterService and TaskTimeoutHandler class names might make it easier to go find the code -- I had to grep using the expected `r.toString()` value below. 

But the comment gives a hint, so I'm content and will leave it up to you.

Reply:
How about this: I extracted the method that generates the string to `org.elasticsearch.cluster.service.MasterService#getTimeoutTaskDescription`, that way you can use `Find usages` to join the dots.